server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -480,4 +481,55 @@ private TranslogDeletionPolicy getTranslogDeletionPolicy(EngineConfig engineConf
         );
     }
 
+    public synchronized void finalizeReplication(SegmentInfos infos, Store.MetadataSnapshot expectedMetadata, long seqNo)
```

Comment:
The intention was for the segrep target to invoke the `updateSegments` method that exists already in this class when a replication event is finalizing, if there are changes can you pls add there?  A lot of this looks like it was what we were doing in the poc inside InternalEngine and is not required here.  I thin maybe the piece missing from the current method is the new call to store.cleanupAndVerify, though I think we can do all that work in the target instead?